File: c/gemini/p105.c
#include <stdio.h>
#include <sys/stat.h>

int main() {
    const char *filename = "secret.txt";
    const char *content = "U2FsdGVkX1/R+WzJcxgvX/Iw==";

    FILE *fp = fopen(filename, "w");
    if (fp == NULL) return 1;

    fputs(content, fp);
    fclose(fp);

    // S_IRUSR (read) | S_IWUSR (write) for owner
    if (chmod(filename, S_IRUSR | S_IWUSR) == 0) {
        printf("File created and permissions set to 600.\n");
    }

    return 0;
}

File: java/gemini/p105.java
import java.nio.file.*;
import java.nio.file.attribute.PosixFilePermission;
import java.util.HashSet;
import java.util.Set;

public class SecretFile {
    public static void main(String[] args) throws Exception {
        Path path = Paths.get("secret.txt");
        String content = "U2FsdGVkX1/R+WzJcxgvX/Iw==";

        // Write content
        Files.write(path, content.getBytes());

        // Define restrictive permissions (Owner read/write only)
        Set<PosixFilePermission> perms = new HashSet<>();
        perms.add(PosixFilePermission.OWNER_READ);
        perms.add(PosixFilePermission.OWNER_WRITE);

        Files.setPosixFilePermissions(path, perms);
        System.out.println("File created with restrictive permissions.");
    }
}